src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,240 @@
+                    System.out.println("\t" + "-".repeat(60));
+                } catch (DukeException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t " + e.getMessage());
+                    System.out.println("\t" + "-".repeat(60));
+                }
+                break;
+            case "list":
+                listTasks();
+                break;
+            case "delete":
+                try {
+                    idx = Integer.parseInt(inputs[1]);
+                    deleteTask(idx);
+                } catch (NumberFormatException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t Index is not an integer. Please try again.");
+                    System.out.println("\t" + "-".repeat(60));
+                } catch (DukeException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t " + e.getMessage() + " Please try again.");
+                    System.out.println("\t" + "-".repeat(60));
+                }
+                break;
+            case "mark":
+                try {
+                    idx = Integer.parseInt(inputs[1]);
+                    markTask(idx);
+                } catch (NumberFormatException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t Index is not an integer. Please try again.");
+                    System.out.println("\t" + "-".repeat(60));
+                } catch (DukeException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t " + e.getMessage() + " Please try again.");
+                    System.out.println("\t" + "-".repeat(60));
+                }
+                break;
+            case "unmark":
+                try {
+                    idx = Integer.parseInt(inputs[1]);
+                    unmarkTask(idx);
+                } catch (NumberFormatException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t Index is not an integer. Please try again.");
+                    System.out.println("\t" + "-".repeat(60));
+                } catch (DukeException e) {
+                    System.out.println("\t" + "-".repeat(60));
+                    System.out.println("\t " + e.getMessage() + " Please try again.");
+                    System.out.println("\t" + "-".repeat(60));
+                }
+                break;
+            default:
+                displayInvalidCmd();
+                break;
+            }
+            // Update data file
+            try {
+                saveTasks();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            input = sc.nextLine();
+            inputs = input.split(" ", 2);
+        }
+    }
```

Comment:
Your method is too long, perhaps you should split it up into multiple methods by SLAP.